cupy/linalg/product.py

```diff
@@ -59,6 +60,127 @@ def vdot(a, b):
     return core.tensordot_core(a, b, None, 1, 1, a.size, ())
 
 
+def cross(a, b, axisa=-1, axisb=-1, axisc=-1, axis=None):
+    """Returns the cross product of two vectors.
+
+    The cross product of `a` and `b` in :math:`R^3` is a vector perpendicular
+    to both `a` and `b`.  If `a` and `b` are arrays of vectors, the vectors
+    are defined by the last axis of `a` and `b` by default, and these axes
+    can have dimensions 2 or 3.  Where the dimension of either `a` or `b` is
+    2, the third component of the input vector is assumed to be zero and the
+    cross product calculated accordingly.  In cases where both input vectors
+    have dimension 2, the z-component of the cross product is returned.
+
+    Args:
+        a : cupy.ndarray
+            Components of the first vector(s).
+        b : cupy.ndarray
+            Components of the second vector(s).
+        axisa : int, optional
+            Axis of `a` that defines the vector(s).  By default, the last axis.
+        axisb : int, optional
+            Axis of `b` that defines the vector(s).  By default, the last axis.
+        axisc : int, optional
+            Axis of `c` containing the cross product vector(s).  Ignored if
+            both input vectors have dimension 2, as the return is scalar.
+            By default, the last axis.
+        axis : int, optional
+            If defined, the axis of `a`, `b` and `c` that defines the vector(s)
+            and cross product(s).  Overrides `axisa`, `axisb` and `axisc`.
+
+    Returns:
+        c : cupy.ndarray
+            Vector cross product(s).
+
+
+    .. seealso:: :func:`numpy.cross`
+
+    """
+
+    if axis is not None:
+        axisa, axisb, axisc = (axis,) * 3
+    a = cupy.asarray(a)
+    b = cupy.asarray(b)
+    # Check axisa and axisb are within bounds
+    axisa = ch.normalize_axis_index(axisa, a.ndim, msg_prefix='axisa')
```

Comment:
I think `normalize_axis_index` is not documented.
Let's avoid using it, or at least add a TODO to avoid using an undocumented function.

Reply:
normalize_axis_index is a function from numpy and it is documented

Reply:
Could you direct me to the documentation?
I can only find this site (apparantly it's not official): https://kite.com/python/docs/numpy.polynomial.chebyshev.normalize_axis_index

Reply:
I found the documentation from python shell
> help(numpy.polynomial.chebyshev.normalize_axis_index)

Reply:
Well, that doesn't mean it's documented.
No one can find it unless he knows the function in the first place.

There's no such function in the official documentation: https://docs.scipy.org/doc/numpy/reference/routines.polynomials.chebyshev.html

Reply:
Probably it's meant to be a hidden function, not exposed to user?

Reply:
If necessary, we can implement this function in CuPy, and document it

Reply:
Yes, but that should be done separately.
I think it's easier to just leave TODO in this PR.

Reply:
Added a TODO just before first call of normalize_axis_index